rasa/core/policies/memoization.py

```diff
@@ -286,28 +287,36 @@ class AugmentedMemoizationPolicy(MemoizationPolicy):
     """
 
     @staticmethod
-    def _back_to_the_future(
+    def _strip_leading_events_until_action_executed(
         tracker: DialogueStateTracker, again: bool = False
     ) -> Optional[DialogueStateTracker]:
-        """Send Marty to the past to get
-        the new featurization for the future"""
+        """Truncates the tracker to begin at the next `ActionExecuted` event.
 
+        Args:
+            tracker: The tracker to truncate.
+            again: When true, truncate tracker at the second action.
+                Otherwise truncate to the first action.
+
+        Returns:
+            The truncated tracker if there were actions present.
+            If none are found, returns `None`.
+        """
         idx_of_first_action = None
         idx_of_second_action = None
 
         applied_events = tracker.applied_events()
 
-        # we need to find second executed action
+        # We need to find the second `ActionExecuted` or `UserUttered` event.
```

Comment:
This is not true. The code below finds the second ActionExecuted, no?

Reply:
Ah forget to change that back. Reverted to original.